File: recyclerlayout/src/main/java/com/angel/adapter/SWViewHolder.java
package com.angel.adapter;

import android.support.v7.widget.RecyclerView;
import android.util.SparseArray;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.ImageView;
import android.widget.TextView;

/**
 * Created by Angel on 2017/4/7.
 */
public class SWViewHolder extends RecyclerView.ViewHolder {

    private SparseArray<View> viewSparseArray;

    public SWViewHolder(View itemView) {
        super(itemView);
        viewSparseArray = new SparseArray<>();
    }

    private <T extends View> T findViewById(int viewId) {
        View view = viewSparseArray.get(viewId);
        if (view == null) {
            view = itemView.findViewById(viewId);
            viewSparseArray.put(viewId, view);
        }
        return (T) view;
    }

    public View getView(int viewId) {
        return findViewById(viewId);
    }

    public TextView getTextView(int viewId) {
        return (TextView) getView(viewId);
    }

    public Button getButton(int viewId) {
        return (Button) getView(viewId);
    }

    public ImageView getImageView(int viewId) {
        return (ImageView) getView(viewId);
    }

    public SWViewHolder setText(int viewId, String value) {
        TextView view = findViewById(viewId);
        view.setText(value);
        return this;
    }
}

File: kotlinlibrary/src/main/java/com/kotlin/adapter/SWViewHolder.kt
package com.kotlin.adapter

import android.support.v7.widget.RecyclerView
import android.util.SparseArray
import android.view.View
import android.widget.Button
import android.widget.ImageView
import android.widget.TextView
import com.kotlin.layout.SWSlipeLayout

/**
 * Created by Angel on 2017/5/19.
 */
class SWViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {

    private val viewSparseArray: SparseArray<View>

    init {
        viewSparseArray = SparseArray<View>()
    }

    private fun <T : View> findViewById(viewId: Int): T {
        var view: View? = viewSparseArray.get(viewId)
        if (view == null) {
            view = itemView.findViewById(viewId)
            viewSparseArray.put(viewId, view)
        }
        return (view as T?)!!
    }

    fun getView(viewId: Int): View {
        return findViewById(viewId)
    }

    fun getTextView(viewId: Int): TextView {
        return getView(viewId) as TextView
    }

    fun getButton(viewId: Int): Button {
        return getView(viewId) as Button
    }

    fun getImageView(viewId: Int): ImageView {
        return getView(viewId) as ImageView
    }

    fun getSWSlipLayout(viewId: Int): SWSlipeLayout {
        return getView(viewId) as SWSlipeLayout
    }
}
